### ISSUE TITLE
NPE in Aero calculateBattleValue

### ISSUE THREAD

**Issue by pheonixstorm** [MegaMek/megamek#52]
Started a new AtB scenario with a pair of aero I used in the previous. Previous scenario I added bombs in the lobby and ended the game with a few left. Didn't even think about it and started up the new scenario. MM loads up the group of mechs fine but glitched when it hit the aero.

Tried to reload the scenario w/o the bombs but got the same result. This time the UI did not glitch though and was able to display all units except the aero.

Restarted HQ, no bombs on aero, restarted scenario. No issues with either aero.

[aeroNPE.txt](https://github.com/MegaMek/megamek/files/154271/aeroNPE.txt)

**Comment by arlith** [comment]
Really would like a saved game for this.  I can see the cause of the NPE, but it's not something I would expect to happen (basically, it appears that there's a Mounted with a null EquipmentType).  I would like to know how things got to this state.  Just fixing the NPE will likely mask a larger issue.

**Comment by pheonixstorm** [comment]
Didn't keep the previous MM save but will try to reproduce from a fresh HQ campaign.

**Comment by pheonixstorm** [comment]
In this test I also received and index out of bounds. It may have been there before but didn't notice due to the endless list of NPEs

[aeroNPE.zip](https://github.com/MegaMek/megamek/files/157706/aeroNPE.zip)

**Comment by arlith** [comment]
What am I supposed to do with the attachment?  I loaded up the saved game but didn't encounter an issue.  Then I loaded up the cpnx, started a game, threw in Princess with a few units, and still didn't see an issue.

**Comment by pheonixstorm** [comment]
The save was the prior battle just to show the unit having bombs. The campaign is current for the next scenario. If you aren't having the issue you may need to start the second scenario and fight a few turns until some bombs are dropped, declare victory and resolve in HQ then start a new scenario and deploy all forces (don't change the number of bombs). Once the lobby comes up you should get the NPE.

It is possible that HQ is sending the wrong information and a save/reload fixes it.

**Comment by arlith** [comment]
Yea, it's likely  going to be a hard issue to reproduce.  What's happening is the Mounted Equipment type is null, which I think only happens when it can't find a matching piece of equipment when adding a new Mounted.  This isn't supposed to happen, so figuring out what's causing it might be tricky...

**Comment by pheonixstorm** [comment]
I have an idea on this. I have not used inferno bombs lately and have not had this NPE pop up again. I am wondering if having, or having used, inferno bombs between sessions could be an issue. Especially considering that the TL does like to reset to standard for some odd reason. Perhaps the empty inferno slot is still being sent and checked? Only other thing I can think of on this.

**Comment by arlith** [comment]
I had a hunch it was related to bombs as well, but still hasn't gotten me closer to getting ti fixed.

**Comment by pheonixstorm** [comment]
Ran into this again. Start scenario, MM loads to the lobby and stops due to NPE with the aero again. Saving and reloading the campaign cleared this up. I suspect the problem is on the HQ side.

I have the most recent files if you want them. Will attempt to reproduce this and write down the steps.

**Comment by Menzius** [comment]
The same issue happened to me I think. I used two ASFs (Eisensturm prime, Riever F-700b) loaded with max payload of rockets. The Riever failed a maneuver and flew off the map while it still had rockets attached and did not return before /victory was declared, it is also marked as an Illegal design. Nothing else out of the ordinary happened. During the next scenario MegaMek glitched at the point the ASFs got transferred. Undeploying the ASFs or a simply reload fixed the problem. As I undeployed them MekHQ showed both ASFs in red as if they were damaged while both are fine, reload fixed this as well. 

Attached are the mekhqlog.txt, the campaign save shoudn't be helpful since the issue is fixed by a reload.
[Aero bug report mekhqlog.txt](https://github.com/MegaMek/megamek/files/427791/Aero.bug.report.mekhqlog.txt)
 About 3/4 of the log are errors.
